accounting/transferwise/tests/test_models.py

```diff
@@ -0,0 +1,153 @@
+    def test_create_payments(self, date):
+        """ Payments are automatically created for all of this bulk payment's recipients. """
+        self._create_recipients(invoices=True, invoice_date=date)
+        self.bulk_payment.create_payments()
+        self.assertEqual(models.TransferWisePayment.objects.all().count(), 2)
+
+    def test_create_payments_which_already_exist(self):
+        """ Payments are not created because they already exist in the system. """
+        self._create_recipients(invoices=True)
+        for unused in range(3):
+            self.bulk_payment.create_payments()
+            self.assertEqual(models.TransferWisePayment.objects.all().count(), 2)
+
+    def test_create_payments_no_recipients(self):
+        """ Payments are not created because no valid recipients exist. """
+        self.bulk_payment.create_payments()
+        self.assertFalse(models.TransferWisePayment.objects.all().exists())
+
+    def test_create_payments_no_invoice(self):
+        """ Payments are not created because an invoice does not exist for the recipients. """
+        self._create_recipients()
+        self.bulk_payment.create_payments()
+        self.assertFalse(models.TransferWisePayment.objects.all().exists())
+
+    def test_csv_filename(self):
+        """ The CSV filename contains the proper sender and date. """
+        self.assertEqual(self.bulk_payment.csv_filename, 'transferwise_bulk_payment_csv_opencraft_2018-01-10.csv')
+
+    @mock.patch.object(models.TransferWiseBulkPayment, 'create_payments')
+    def test_auto_create_payments_on_save(self, mock_create_payments):
+        """ `create_payments` is called when `auto_create_payments_on_save` is checked. """
+        self.bulk_payment.auto_create_payments_on_save = True
+        self.bulk_payment.save()
+        mock_create_payments.assert_called_once()
+
+    @mock.patch.object(models.TransferWiseBulkPayment, 'to_bulk_payment_csv')
+    def test_auto_create_csv_on_save(self, mock_to_bulk_payment_csv):
+        """ `to_bulk_payment_csv` is called when `auto_create_csv_on_save` is checked. """
+        self.bulk_payment.auto_create_csv_on_save = True
+        self.bulk_payment.save()
+        mock_to_bulk_payment_csv.assert_called_once()
+
+    @mock.patch.object(models.TransferWiseBulkPayment, 'upload_to_google_drive')
+    def test_auto_upload_google_drive_on_save(self, mock_upload_to_google_drive):
+        """ `upload_to_google_drive` is called when `auto_upload_google_drive_on_save` is checked. """
+        csv_path = 'https://drive.google.com/bulk_payment.csv'
+        mock_upload_to_google_drive.return_value = {'webContentLink': csv_path}
+        self.bulk_payment.auto_upload_google_drive_on_save = True
+        self.bulk_payment.save()
+        mock_upload_to_google_drive.assert_called_once()
+        self.assertEqual(self.bulk_payment.csv_path, csv_path)
+
+    def _create_recipients(self, invoices=False, invoice_date=NOW):
```

Comment:
@UmanShahzad Also, let's call the second parameter `create_invoices` for clarity.

Reply:
:+1: